src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,217 @@
+  const [startClientX, setStartClientX] = useState<number>(0)
+  const [endClientX, setEndClientX] = useState<number>(0)
+  const [cursorOn, setCursorOn] = useState<boolean>(false)
+
+  const handleMoveCurrent = (imageIndex: number): void => {
+    setTranslateValue((imageIndex - 1) * carouselWidthSize)
+  }
+  const handleMoveRight = (): void => {
+    if (translateValue !== carouselWidthSize * (images.length - 1)) {
+      setTranslateValue(prev => prev + carouselWidthSize)
+    } else {
+      setTranslateValue(0)
+    }
+  }
+
+  const handleMoveLeft = (): void => {
+    if (translateValue !== 0) {
+      setTranslateValue(prev => prev - carouselWidthSize)
+    } else {
+      setTranslateValue(carouselWidthSize * (images.length - 1))
+    }
+  }
+
+  const handleMouseDown = (e: { clientX: SetStateAction<number> }): void => {
+    setStartClientX(e.clientX)
+    setCursorOn(true)
+  }
+
+  const handleMouseUp = (e: { clientX: SetStateAction<number> }): void => {
+    setEndClientX(e.clientX)
+    setCursorOn(false)
+  }
+  const handleTouchStart: TouchEventHandler<HTMLDivElement> = e => {
+    setEndClientX(e.touches[0].clientX)
+    setCursorOn(true)
+  }
+  const handleTouchStartEnd: TouchEventHandler<HTMLDivElement> = e => {
+    setStartClientX(e.changedTouches[0].clientX)
+    setCursorOn(false)
+  }
+
+  useEffect(() => {
+    const dragSpace = Math.abs(startClientX - endClientX)
+    const userSlideRight = endClientX < startClientX && dragSpace > 100
+    const userSlideLeft = endClientX > startClientX && dragSpace > 100
+    if (startClientX === 0) {
+      return
+    }
+    if (userSlideRight) {
+      handleMoveRight()
+    } else if (userSlideLeft) {
+      handleMoveLeft()
+    }
+  }, [endClientX])
+
+  return (
+    <StyledCarouselWrapper>
+      <StyledSlider
+        cursorOn={cursorOn}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchEnd={handleTouchStartEnd}
+        onTouchStart={handleTouchStart}>
+        <StyledImageBox translateValue={translateValue || null}>
+          {images.map((image, idx) => {
+            return (
+              <StyledImage key={image.id} alt={'image' + idx} src={image.url} />
+            )
+          })}
+        </StyledImageBox>
+        {isArrow && (
+          <StyledArrowBox>
+            <StyledArrow
+              alt="arrow-left"
+              src={arrowLeft}
+              onClick={handleMoveLeft}></StyledArrow>
+            <StyledArrow
+              alt="arrow-right"
+              src={arrowRight}
+              onClick={handleMoveRight}></StyledArrow>
+          </StyledArrowBox>
+        )}
+      </StyledSlider>
+      <StyledDotBox>
+        {images.map(image => {
+          return (
+            <StyledDot
+              key={image.id}
+              className={`${image.id}`}
+              onClick={(): void => {
+                handleMoveCurrent(image.id)
+              }}></StyledDot>
```

Comment:
요기도 싱글 태그로 변경하면 좋겠네요 !
onClick에 함수를 적용할 때 함수를 바로 넣어도 괜찮을 것 같습니다!
이전에 name을 받게 된다면, 혹시나하는 여러개의 케러셀을 한 페이지에서 공유하는 경우 class의 값이 중복되지 않도록 
사용할 수 있을 것 같습니다.
```suggestion
            <StyledDot
              key={image.id}
              className={`${name}-${image.id}`}
              onClick={handleMoveCurrent(image.id)}} />
```

Reply:
`handleMoveCurrent(image.id)`
이렇게 걸어주면 이벤트 바인딩할 때, 함수를 계속 실행시켜서 무한 렌더링되지 않나용+_+??

Reply:
<img width="765" alt="image" src="https://user-images.githubusercontent.com/47546413/188310598-d2ccc7c6-3e1b-4e10-8c37-aeb83916c4a6.png">

<img width="713" alt="image" src="https://user-images.githubusercontent.com/47546413/188310610-caa308c1-150a-4776-a5ea-604477a3e9f9.png">

무한 렌더링, lint 에러 이슈가 있는 것 같슴니다! 🥲

Reply:
오호 그렇군요 ! 정확하게 몰랐는데 알게됐네요 

확인해주셔서 감사합니다 ㅎㅎ 하나 알아가네요 🙇‍♀️